src/state/dimension.js

```diff
@@ -136,8 +143,12 @@ export const getDroppableDimension = ({
 }: GetDroppableArgs): DroppableDimension => {
   const withMargin = getWithSpacing(clientRect, margin);
   const withWindowScroll = getWithPosition(clientRect, windowScroll);
-  // If no containerRect is provided, the clientRect is the container
-  const containerRectWithWindowScroll = getWithPosition(containerRect || clientRect, windowScroll);
+  // If no containerRect is provided, or if the clientRect matches the containerRect, this
+  // droppable is its own container. In that case we include margin in the container dimension.
+  const droppableIsContainer = !containerRect || equal(containerRect, clientRect);
+  const containerRectWithWindowScroll = droppableIsContainer
```

Comment:
line 174 (page > withoutMargin) should be renamed now - sometimes it includes margins and sometimes it doesn't...